lua/weapons/weapon_airgun.lua

```diff
@@ -0,0 +1,120 @@
+
+SWEP.PrintName	= "Air Gun"
+SWEP.Category = "Other"
+SWEP.Author	= "Austin and Noz"
+SWEP.Instructions = "Push stuff"
+
+local ForcePower = 300 -- change it if u want change air-power
+
+SWEP.DrawCrosshair		= false
+SWEP.Primary.ClipSize		 = -1
+SWEP.Primary.DefaultClip	 = -1
+SWEP.Primary.Automatic		= false
+SWEP.Primary.Ammo		 = "none"
+SWEP.Primary.Delay = 1.25
+
+SWEP.Secondary.ClipSize	  = -1
+SWEP.Secondary.DefaultClip  = -1
+SWEP.Secondary.Automatic	 = false
+SWEP.Secondary.Ammo	  = "none"
+
+SWEP.Spawnable = true
+SWEP.ViewModel = "models/milaco/airgun/airgun.mdl"
+SWEP.WorldModel = "models/milaco/airgun/airgun.mdl"
+
+function SWEP:PrimaryAttack()
+	
+	self.Owner:ExtEmitSound("airgun/air.wav")
+	
+	self.Weapon:SetNextPrimaryFire(CurTime() + self.Primary.Delay)
+	
+	local tr = self:GetOwner():GetEyeTrace(MASK_SHOT)
+	
+	if tr.Hit and IsValid(tr.Entity) and (self:GetOwner():EyePos() - tr.HitPos):Length() < 100 and not Safe(tr.Entity) then
+	
+		local enthit = tr.Entity
+		if enthit:IsRagdoll() or enthit:GetClass() == "prop_physics" then
+		
+			local phys = enthit:GetPhysicsObject()
+			if IsValid(phys) and phys:IsMoveable() then
+			
+				local pdir = tr.Normal
+				local speed = phys:GetVelocity():Length()
+				local maxforce = 9000
+				local force = (maxforce + (10 - maxforce) * (speed / 125)) * ForcePower/100
+				
+				if enthit:GetClass() == "prop_ragdoll" then

```

Comment:
If ragdolls seemed too slow it's because they have many physics objects, one for each bone, and the ragdoll's effective mass is the sum of all of them, but you're only looking at one bone here. We don't use serverside ragdolls so whatever.